ha/core/event_analyzer/parser/parser.py

```diff
@@ -178,33 +179,38 @@ def parse_event(self, msg: str) -> HealthEvent:
         try:
             message = json.dumps(ast.literal_eval(msg))
             cluster_resource_alert = json.loads(message)
-            timestamp = str(int(time.time()))
-            event_id = timestamp + str(uuid.uuid4().hex)
-            node_id = cluster_resource_alert["_resource_name"]
-            resource_type = cluster_resource_alert["_resource_type"]
-            event_type = cluster_resource_alert["_event_type"]
-            timestamp = cluster_resource_alert["_timestamp"]
-            generation_id = cluster_resource_alert["_generation_id"]
+            timestamp = cluster_resource_alert["event"][EVENT_ATTRIBUTES.HEALTH_EVENT_HEADER.value][HEALTH_ATTRIBUTES.TIMESTAMP.value]
+            event_id = cluster_resource_alert["event"][EVENT_ATTRIBUTES.HEALTH_EVENT_HEADER.value][HEALTH_ATTRIBUTES.EVENT_ID.value]
+            source = cluster_resource_alert["event"][EVENT_ATTRIBUTES.HEALTH_EVENT_PAYLOAD.value][HEALTH_ATTRIBUTES.SOURCE.value]
+            node_id = cluster_resource_alert["event"][EVENT_ATTRIBUTES.HEALTH_EVENT_PAYLOAD.value][HEALTH_ATTRIBUTES.NODE_ID.value]
+            resource_type = cluster_resource_alert["event"][EVENT_ATTRIBUTES.HEALTH_EVENT_PAYLOAD.value][HEALTH_ATTRIBUTES.RESOURCE_TYPE.value]
+            resource_id = cluster_resource_alert["event"][EVENT_ATTRIBUTES.HEALTH_EVENT_PAYLOAD.value][HEALTH_ATTRIBUTES.RESOURCE_ID.value]
+            event_type = cluster_resource_alert["event"][EVENT_ATTRIBUTES.HEALTH_EVENT_PAYLOAD.value][HEALTH_ATTRIBUTES.RESOURCE_STATUS.value]
+            specific_info = cluster_resource_alert["event"][EVENT_ATTRIBUTES.HEALTH_EVENT_PAYLOAD.value][HEALTH_ATTRIBUTES.SPECIFIC_INFO.value]
+            if specific_info and specific_info["generation_id"] and source == "monitor":
```

Comment:
We already have HEALTH_EVENT_SOURCES. Can't we use the MONITOR defined there ? Or have another define for the same ?

Reply:
Ohh Yes.. Missed this. Thanks. Fixed now